### PR TITLE
Don't include deprecated rules in `all` config

### PR DESCRIPTION
**Pull request by pfhayes**
Since they are replaced by better versions

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -39,7 +39,9 @@ describe('configurations', function() {
       assert.equal(plugin.configs.all.rules[configName], 2);
     });
     rules.forEach(function(ruleName) {
-      assert(plugin.configs.all.rules['react/' + ruleName]);
+      if (!plugin.deprecatedRules[ruleName]) {
```
Comment (ljharb) [inline tests/index.js]: we should assert instead that every rule is either in `deprecatedRules` or `all`, rather than just skipping the deprecated ones.

**Comment by ljharb** [comment]
Good catch!

File: tests/index.js (hunk 1 above)
Comment (pfhayes) [inline tests/index.js]: good call, I'll make that more explicit

File: tests/index.js [hunk 2]
```diff
@@ -38,8 +38,10 @@ describe('configurations', function() {
       assert.equal(configName.indexOf('react/'), 0);
       assert.equal(plugin.configs.all.rules[configName], 2);
     });
-    rules.forEach(function(ruleName) {
-      assert(plugin.configs.all.rules['react/' + ruleName]);
+    ruleFiles.forEach(function(ruleName) {
+      var inDeprecatedRules = Boolean(plugin.deprecatedRules[ruleName]);
```
Comment (ljharb) [inline tests/index.js]: fwiw `!!foo` and `Boolean(foo)` are equivalent, except that the former is faster, and can't be forged

Comment (pfhayes) [inline tests/index.js]: I'm happy to change, but it looked like in this project `Boolean` (12 occurrences) was preferred over `!!` (0 occurrences)

Comment (ljharb) [inline tests/index.js]: Sounds good, keep it as `Boolean` for now then. We can change it to `!!` separately, if desired.

**Comment by yannickcr** [comment]
Merged, thanks!

**Pull request merged**